.github/workflows/workflow.yaml

```diff
@@ -53,6 +57,18 @@ jobs:
           for dir in *; do
```

Comment:
I am worried about the increasingly nested, loop, if-else driven structure of this script.
Perhaps we can think about improving it's readability.
For example:
```
...
bazel_run_with_nix() {
  echo Running $(head -n1 README.md) with Nix
  nix-shell --command 'bazel run --config=nix :hello'
  cd ..
}

bazel_run_without_nix() {
  echo Running $(head -n1 README.md) without Nix
  bazel run :hello
  cd ..
}

bazel_skip() {
  echo Skipping $(head -n1 README.md)
  cd ..
}

runner_os="${{ runner.os }}" 
for dir in *; do
  cd "${dir}"
  echo
  case "${dir},${runner_os}" in
    python,macOS)
      bazel_skip
      continue
      ;;
    rust,macOS)
      bazel_skip
      continue
      ;;
    cc,*)
      bazel_run_with_nix
      bazel_run_without_nix
      ;;
    go,*)
      bazel_run_with_nix
      bazel_run_without_nix
      ;;
    rust,*)
      bazel_run_with_nix
      bazel_run_without_nix
      ;;
    *,*)
      bazel_run_with_nix
      ;;
  esac
done
```

Reply:
I'm more afraid of building a complicated bash-based test matrix runner than just nesting and if. Maybe I could make it a bit simpler? I'll try.

Reply:
Please check out the new version - it seems even more straightforward.